pkg/alerting/alert_handler.go

```diff
@@ -6,24 +6,43 @@ import (
 	"net/http"
 	"time"
 
+	"go.uber.org/fx"
+	"go.uber.org/zap"
+	"golang.org/x/exp/maps"
+
 	"github.com/uptrace/bun"
 	"github.com/uptrace/bunrouter"
+	"github.com/uptrace/go-clickhouse/ch"
+	"github.com/uptrace/opentelemetry-go-extra/otelzap"
 	"github.com/uptrace/uptrace/pkg/attrkey"
 	"github.com/uptrace/uptrace/pkg/bunapp"
+	"github.com/uptrace/uptrace/pkg/bunconf"
 	"github.com/uptrace/uptrace/pkg/httperror"
 	"github.com/uptrace/uptrace/pkg/httputil"
 	"github.com/uptrace/uptrace/pkg/org"
-	"go.uber.org/zap"
-	"golang.org/x/exp/maps"
 )
 
+type AlertHandlerParams struct {
+	fx.In
+	Logger *otelzap.Logger
+	Conf   *bunconf.Config
+	PG     *bun.DB
+	CH     *ch.DB
+}
+
 type AlertHandler struct {
-	*bunapp.App
+	logger *otelzap.Logger
+	conf   *bunconf.Config
+	pg     *bun.DB
+	ch     *ch.DB
 }
 
-func NewAlertHandler(app *bunapp.App) *AlertHandler {
+func NewAlertHandler(p AlertHandlerParams) *AlertHandler {
 	return &AlertHandler{
-		App: app,
+		logger: p.Logger,
+		conf:   p.Conf,
+		pg:     p.PG,
+		ch:     p.CH,
```

Comment:
It looks like for every handler we need to repeat each dependency/param 3 times. It might be worth it in some complex cases when we need to turns params into something else, but with handlers it just looks like a waste of time.

Can't we just embed params in the handler?

```go
type AlertHandler struct {
	AlertHandlerParams
	*bunapp.App
}
```